base-notebook/start-notebook.sh

```diff
@@ -19,4 +19,4 @@ if [[ "${RESTARTABLE}" == "yes" ]]; then
 fi
 
 # shellcheck disable=SC1091,SC2086
-exec /usr/local/bin/start.sh ${wrapper} jupyter ${DOCKER_STACKS_JUPYTER_CMD} ${NOTEBOOK_ARGS} "$@"
+exec /usr/local/bin/start.sh ${wrapper} jupyter ${DOCKER_STACKS_JUPYTER_CMD} --port ${JUPYTER_PORT} ${NOTEBOOK_ARGS} "$@"
```

Comment:
`JUPYTER_PORT` env variable is no longer considered as a port for jupyter applications, so we pass this explicitly - I think the easiest way to do this is to add this line to both `start-notebook.sh` and `start.sh`.

We, on the other hand, can and should keep the variable name `JUPYTER_PORT` for users of our docker image.
This way we won't break them (unless they call `start.sh` manually, but I support it's a rare case).

Reply:
Normally this should still be supported I think?

![image](https://github.com/jupyter/docker-stacks/assets/591645/c0dff377-b9ae-43ec-bac8-30d59aeb406b)

Reply:
Since it's part of Jupyter Server now: https://github.com/jupyter-server/jupyter_server/blob/c1d689e05169ba47e630ccaa39cebb4f847b3562/jupyter_server/serverapp.py#L950

Reply:
Thanks, I will try to revert this part.